src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -52,15 +52,20 @@ public void deploy(DeployConfiguration config) throws AppEngineException {
     Preconditions.checkNotNull(config);
     Preconditions.checkNotNull(config.getDeployables());
     Preconditions.checkArgument(config.getDeployables().size() > 0);
+    File workingDirectory = null;
 
     List<String> arguments = new ArrayList<>();
     arguments.add("deploy");
-    for (File deployable : config.getDeployables()) {
-      if (!deployable.exists()) {
-        throw new IllegalArgumentException(
-            "Deployable " + deployable.toPath() + " does not exist.");
+    if (config.getDeployables().size() == 1 && config.getDeployables().get(0).isDirectory()) {
```

Comment:
Can you preface this line with a comment explaining the purpose of this conditional?

It's a kind of strange case when gcloud app deploy is being run in a java directory, and even stranger that you can't actually specify a directory from glcoud to do this (that it must be done in current working directory).